src/bundle.c

```diff
@@ -855,10 +856,20 @@ static int install_bundles(struct list *bundles, struct list **subs, int current
 	run_scripts(false);
 
 	ret = 0;
-	fprintf(stderr, "Bundle(s) installation done.\n");
 	print_time_stats(&times);
 
 out:
+	bundles_installed = list_len(to_install_bundles);
+	bundles_failed = bundles_requested - bundles_installed;
+	/* the bundles installed could end up being more than the ones requested
+	 * in certain situations like when a bundle needs to be updated */
+	if (bundles_failed > 0) {
```

Comment:
I don't like the error output when you try to install a bundle that is already present.

    # swupd bundle-add os-core
    Warning: Bundle "os-core" is already installed, skipping it...
    1 bundle of 1 failed to be installed

Can't we output something like: "1 bundle was already installed". What do you think about that?

Reply:
Ok, I can do that. In this case, what do you think the exit status should be? 0? Or we should stick to an exit code >0?

Reply:
I think the best option is to return an error code even when trying to install a bundle that is already installed. This way it will also be more consistent with the way bundle-remove works. The only way the user will get exit code = 0 is if everything he requested is installed/removed successfully.